src/file/reader.rs

```diff
@@ -515,4 +558,24 @@ mod tests {
     assert!(file.is_ok());
     file.unwrap()
   }
+
+  fn get_temp_file<'a>(file_name: &str, content: &[u8]) -> fs::File {
+    // build tmp path to a file in "target/debug/testdata"
+    let mut path_buf = env::current_dir().unwrap();
+    path_buf.push("target");
+    path_buf.push("debug");
+    path_buf.push("testdata");
+    fs::create_dir_all(&path_buf).unwrap();
```

Comment:
Wonder if we need to take care of the case when a path already exist (e.g., calling `get_temp_file` twice in the same fn). Seems in that case it will return `Err`?

Reply:
I tried calling this method multiple times on the same directory - it is no-op, it is similar to mkdir -p. Plus, I run tests several times and it works! I think it should be okay.

Reply:
Sounds good!